src/modules/r.rs

```diff
@@ -0,0 +1,87 @@
+use regex::Regex;
+
+use super::{Context, Module, RootModuleConfig};
+
+use crate::configs::r::RConfig;
+use crate::formatter::StringFormatter;
+use crate::utils;
+
+const R_VERSION_PATTERN: &str = r" (?P<rversion>\d+\.\d+\.\d+) ";
+
+/// Creates a module with the current Node.js version
+///
+/// Will display the Node.js version if any of the following criteria are met:
+///     - Current directory contains a `.js` file
+///     - Current directory contains a `package.json` or `.node-version` file
+///     - Current directory contains a `node_modules` directory
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let is_r_project = context.try_begin_scan()?.set_extensions(&["R"]).is_match();
+    if !is_r_project {
+        return None;
+    }
+
+    let r_version = utils::exec_cmd("r", &["--version"])?.stderr;
+    let formatted_version = parse_version(&r_version)?;
+    let mut module = context.new_module("r");
+    let config: RConfig = RConfig::try_load(module.config);
+    let formatter = if let Ok(formatter) = StringFormatter::new(config.format) {
+        formatter.map(|variable| match variable {
+            "version" => Some(formatted_version.clone()),
+            _ => None,
+        })
+    } else {
+        log::warn!("Error parsing format string in `r.format`");
+        return None;
+    };
+    module.set_segments(formatter.parse(None));
+    module.get_prefix().set_value("");
+    module.get_suffix().set_value("");
+
+    Some(module)
+}
+
+fn parse_version(version: &str) -> Option<String> {
+    let version_regex = Regex::new(R_VERSION_PATTERN).ok()?;
+    let captures = version_regex.captures(version)?;
+    let r_version = captures["rversion"].to_owned();
+    let r_formatted = format!("{}{}", "v", r_version);
```

Comment:
That seems a bit counter intuitive as usually, the variables get into the right side and static string on the left side. At least from the programming languages I have experience with but it seems the Rust documentation's examples show the same practices.

Maybe that project takes as different approach though? I can change it if that's the case, I don't mind.

Reply:
My suggestion was simply to rather than have the format string, a static string and the variable just have a format string and the variable. Since the static string is not going to change we can simplify this to make it a bit easier to read but if it's not idiomatic rust maybe best to leave it the way it is.

Reply:
Agree with andytom. Though this may not have effect on performance, it improves readability.

Reply:
Hmm I must have been tired (or drunk 🍺 ) when I read the initial suggestion. Sorry @andytom , what you suggested is the idiomatic way. Even the Rust documentation I looked at indicates that (and I didn't leave the link as well... 🤷‍♂️ ). I probably mistaken the before and after. Changing this right now.